addons/src/ao_bookmarklist.c

```diff
@@ -401,14 +409,17 @@ void ao_bookmark_list_activate(AoBookmarkList *bm)
 }
 
 
-void ao_bookmark_list_update(AoBookmarkList *bm, GeanyDocument *doc)
+static gboolean update_bookmark_list_delayed(gpointer data)
 {
 	gint line_nr = 0;
 	gint mask = 1 << 1;
-	ScintillaObject *sci = doc->editor->sci;
+	AoBookmarkListRefreshContainer *container = data;
+	AoBookmarkList *bm = container->bm;
 	AoBookmarkListPrivate *priv = AO_BOOKMARK_LIST_GET_PRIVATE(bm);
+	GeanyDocument *doc = document_find_by_id(container->document_id);
+	ScintillaObject *sci = doc->editor->sci;
```

Comment:
I assume the fix is this?

GeanyDocument *doc = document_find_by_id(container->document_id);

	if (doc != NULL && priv->enable_bookmarklist)
	{
		ScintillaObject *sci = doc->editor->sci;
// etc.